server/src/main/java/org/opensearch/search/pipeline/PipelineProcessingContext.java

```diff
@@ -35,4 +41,31 @@ public void setAttribute(String name, Object value) {
     public Object getAttribute(String name) {
         return attributes.get(name);
     }
+
+    /**
+     * Add a ProcessorExecutionDetail to the list of execution details.
+     *
+     * @param detail the ProcessorExecutionDetail to add
+     */
+    @SuppressWarnings("unchecked")
+    public void addProcessorExecutionDetail(ProcessorExecutionDetail detail) {
+        attributes.computeIfAbsent(PROCESSOR_EXECUTION_DETAILS_KEY, k -> new ArrayList<ProcessorExecutionDetail>());
+        List<ProcessorExecutionDetail> details = (List<ProcessorExecutionDetail>) attributes.get(PROCESSOR_EXECUTION_DETAILS_KEY);
+        details.add(detail);
+    }
```

Comment:
Few extra cycles here for accessing map, how about this version:
```
public void addProcessorExecutionDetail(ProcessorExecutionDetail detail) {
    @SuppressWarnings("unchecked")
    List<ProcessorExecutionDetail> details = (List<ProcessorExecutionDetail>) attributes.computeIfAbsent(
        PROCESSOR_EXECUTION_DETAILS_KEY,
        k -> new ArrayList<>()
    );
    details.add(detail);
}
```

Reply:
Will change in the next revision